tests/compiler/test_slots.py

```diff
@@ -90,8 +95,63 @@ def test_slots_rt():
 
 
 def test_slots_ast():
-    assert_slotted(Location("", 0))
-    assert_slotted(Range("", 0, 0, 0, 0))
+    """
+    Verify that all AST nodes below RequiresEmitStatement and all location objects use slots.
+    """
+    for ast_node_cls in chain(get_all_subclasses(RequiresEmitStatement), get_all_subclasses(Location)):
```

Comment:
nice!

Reply:
It took some puzzling but I think it was worth it.